buildSrc/src/main/groovy/org/elasticsearch/gradle/test/TestWithSslPlugin.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.test;
+
+import org.elasticsearch.gradle.ExportElasticsearchBuildResourcesTask;
+import org.elasticsearch.gradle.precommit.ForbiddenPatternsTask;
+import org.elasticsearch.gradle.testclusters.ElasticsearchCluster;
+import org.elasticsearch.gradle.testclusters.TestClustersPlugin;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
+import org.gradle.api.tasks.TaskProvider;
+
+import java.io.File;
+
+public class TestWithSslPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        File keyStoreDir = new File(project.getBuildDir(), "keystore");
+        TaskProvider<ExportElasticsearchBuildResourcesTask> exportKeyStore = project.getTasks()
+            .register("copyTestCertificates", ExportElasticsearchBuildResourcesTask.class, (t) -> {
```

Comment:
Alternatively, We could reuse the `buildResources` task. I have no strong opinion but my thinking of using a new task here is to detangle this logic from other uses of  `buildResources` within the elasticsearch build logic.

Reply:
Yes. In fact, @rjernst's recent work has _removed_ that global task in favor of individual per-plugin tasks so this is the correct approach going forward.